src/transforms/tree.js

```diff
@@ -43,7 +43,14 @@ export function treeNode({
         layout(root);
         for (const node of root.descendants()) {
           treeFacet.push(++treeIndex);
-          treeData[treeIndex] = node.data;
+          treeData[treeIndex] = {
+            data: node.data,
+            name: nodeName(node),
+            path: nodePath(node),
+            internal: nodeInternal(node),
+            depth: node.depth, // nodeDepth(node)
+            height: node.height // nodeHeight(node)
+          };
```

Comment:
This breaks backwards compatibility, and I think we want to preserve the original data (which is why I originally designed it this way).

Can we not use the existing *node:internal* method because that only works for channels that are visible to the tree transform? I would think that we could pass `filter: "node:internal"` as an option to the treeNode transform? (And we could have a *node:external* or *node:leaf* for the opposite meaning.) But maybe we have to pull out the filter option and do something special; I haven’t tested yet.

Reply:
I couldn't find how to filter inside the transform; the naive approach removes those nodes too early, even before the determination of the tree—basically crashing with a "missing root". There's certainly a better way.

Reply:
I guess the confusion is whether the filter applies before or after the tree is constructed? I guess the filter operation should run before the tree layout… 🤔 Maybe we could have a **treeFilter** option that runs after the tree layout, and supports the same *node:internal* method?